server/src/main/java/org/opensearch/index/mapper/DerivedField.java

```diff
@@ -40,20 +44,47 @@ public DerivedField(StreamInput in) throws IOException {
         name = in.readString();
         type = in.readString();
         script = new Script(in);
+        if (in.readBoolean()) {
+            properties = in.readMap(StreamInput::readString, StreamInput::readString);
+        }
+        sourceIndexedField = in.readOptionalString();
+        format = in.readOptionalString();
+        ignoreMalformed = in.readOptionalBoolean();
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         out.writeString(name);
         out.writeString(type);
         script.writeTo(out);
+        if (properties == null) {
```

Comment:
I think I should also check version here before adding a boolean flag to `StreamOutput` for backward compatibility? As previous version doesn't have the `properties` and other fields introduced here.